specs/bep-v1.rst

```diff
@@ -844,6 +862,142 @@ XDR
+    /                                                               /
+    \                  Name (length + padded data)                  \
+    /                                                               /
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /                                                               /
+    \                      Version Structure                        \
+    /                                                               /
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                    Number of Block Indexes                    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /                                                               /
+    |                    Block Indexes (n items)                    |
+    /                                                               /
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+Fields (DownloadProgress Message)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+**Folder** represents the ID of the folder for which the update is being
+provided.
+
+The **Flags** field is reserved for future use and MUST currently be set to
+zero. The **Options** field contains a list of options that apply to the update.
+
+Fields (FileDownloadProgressUpdate Structure)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The **Update Type** field is made up of the following single bit flags:
+::
+
+     0                   1                   2                   3
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                           Reserved                          |F|
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+:Bit 31 ("F", Forget): is set to notify that the file that was previously
+    advertised is no longer available (as a temporary file).
+
+The **Name** field defines the file name from the global index for which this
+update is being sent.
+
+The **Version** structure defines the version of the file for which this update
+is being sent.
+
+**Block Indexes** is a list of positive integers, where each integer represents
+the index of the block in the FileInfo structure Blocks array that has become
+available for download.
+For example an integer with with value 3 represents that the data defined in the
+fourth BlockInfo structure of the FileInfo structure of that file is now available.
+Please note that matching should be done on **Name** AND **Version**.
+Furthermore, each update received is incremental, for example the initial 
+FileDownloadProgressUpdate structure might contain indexes 0, 1, 2, an update 
+5 seconds later might contain indexes 3, 4, 5 which should be appended to the 
+original list, which implies that blocks 0-5 are currently available.
+
+Block indexes MAY NOT grow in any defined order.
```

Comment:
This forbids the block list to grow in any kind of defined order. :) I think we mean `Block indexes MAY be added in any order. An implementation MUST NOT assume that block indexes are added in any specific order.`